### DISCUSSION TITLE
Urgent window hint when opening a saved session

### DISCUSSION THREAD

**Discussion by cmvanb** [qutebrowser/qutebrowser#8235] [Q&A]
Hi all, Hi Compiler,

I am experiencing one of those bugs, which, although entirely cosmetic, is nevertheless driving me up the wall.

Every time I load a qutebrowser session with a script:

```
qutebrowser --target window ":session-load $session"
```

My wayland compositor, River, receives an urgent window hint from one of the other running qutebrowser instances. Urgent is to be interpreted as needing immediate attention, so I check the other window, but nothing urgent has happened. No crash, no visible errors and qutebrowser has successfully loaded the desired session.

I am trying to find the root cause of the issue, but I am struggling to debug qutebrowser.

I checked `:messages`, all I can see is this warning:
```
17:49:55
WARNING 
qt	
Unknown module:none:0
The cached device pixel ratio value was stale on window update.  Please file a QTBUG which explains how to reproduce.
```

Is that the problem? Some QT/wayland bug? I found this, resolved, QT bug report: https://bugreports.qt.io/browse/QTBUG-117762

I can't seem to find much in the journal or in my compositors log file.

Is there a qutebrowser log file produced somewhere I can investigate?

Any help appreciated.

edit:

I also see this warning:
```
17:59:53
WARNING 
init	
mainwindow:_load_geometry:466
Error while loading geometry.
```

Do warnings trigger the urgent window hint?

**Comment by The-Compiler** [comment]
I *think* this is the same underlying issue as this:

- #5094

As for why it is marked as urgent, it looks like I considered it a good idea some 10 years ago?

- #185

Personally I still find this useful when opening a new link externally to find where it was opened and jump to it quickly, but (just as raising the window) I agree it shouldn't be done for every IPC command.

**Reply by cmvanb** [reply]
It does sound plausible that there is something happening in the window focus logic... I read that issue#5094 and saw @toofar's request for comment, but I don't know how to relate it to the urgency hint.

Does 'raising' a window mean the same as 'focusing' it?

**Comment by cmvanb** [comment]
I did some more testing and found I can reproduce this without any warnings.

The same command as above `qutebrowser --target window ":session-load $session"` will put an urgent window hint on a different qutebrowser window every time.
